### PR TITLE
CRUD of tasks

### PR DESCRIPTION
**Pull request by jlmjkfd**
#39 #41 
edit and delete function of #35 
After the task is manipulated, the page needs to be refreshed manually

### CODE REVIEW

**Comment by ryanwooa** [comment]
1. After added a task, I found that the page will not change, new task will not show in the page. I test in project page and inbox page. I need to refresh the page then the new task will be displayed.
2. I tested updateTask, similar problem. This time, after I editing the content or title, the content where I choose to change is changed. But if I change to other page, the content is not changed. Like I edit a task in today page, which also shown in inbox page. After I make change in the today page and click to the inbox page, the content is not changed. If I back to today page, the content still the old one. Only when I refresh the page, the content will change.

**Comment by jlmjkfd** [comment]
> 1. After added a task, I found that the page will not change, new task will not show in the page. I test in project page and inbox page. I need to refresh the page then the new task will be displayed.
> 2. I tested updateTask, similar problem. This time, after I editing the content or title, the content where I choose to change is changed. But if I change to other page, the content is not changed. Like I edit a task in today page, which also shown in inbox page. After I make change in the today page and click to the inbox page, the content is not changed. If I back to today page, the content still the old one. Only when I refresh the page, the content will change.

Yes, the auto page refresh is not in my plan at the moment, so I mentioned the page needs to be refreshed manually. We need to find an appropriate way to reload the route. For now, the shown task data only loads once when the page.js in the server is loaded, and all the pages in the application are based on this data. That's the reason for your second problem. If we reload the route each time when the data is updated, it will cause a 0.5 to 1-second delay, which will make a bad user experience.

File: components/application/widgets/SingleItem.js [hunk 1]
```diff
@@ -62,132 +78,228 @@ export function SingleItems({
...
-            <div className={styles.task_info}>
-              {hasDue && (<button className={styles.task_info_date} onClick={showScheduler}>
-                <Icon type="calender_small" />
-                {selectedDate}
-              </button>)}
-              {tags.map((tag) => (
-                <Link href={`/application/label/${tag}`}>
-                  <Icon type="small_tag" />
-                  {tag}
-                </Link>
```
Comment (ryanwooa) [inline components/application/widgets/SingleItem.js:92]: Link of tag has been removed?

**Review by kesesek — APPROVED** [review]

File: app/api/addTask/route.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    userId: param.userId,
+    username: param.username,
+    completed: false,
+  });
+
+  try {
+    const res = await newtask
+      .save()
+      .then((savedData) => {
+        console.log("新增成功", savedData);
```
Comment (zli213) [inline app/api/addTask/route.js:30]: Delete or comment this one

File: app/api/addTask/route.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      })
+      .catch((error) => {
+        console.error("新增失败", error);
```
Comment (zli213) [inline app/api/addTask/route.js:33]: Delete or comment this one

**Review by zli213 — APPROVED** [review]
Delete or comment console.log.

**Review by ryanwooa — APPROVED** [review]

**Pull request merged**